src/remote.cc

```diff
@@ -537,18 +537,31 @@ String get_user_name()
     return getenv("USER");
 }
 
+String session_path(StringView session)
+{
+    auto slash_count = std::count(session.begin(), session.end(), '/');
+    if (slash_count > 0)
+        throw runtime_error{"session names cannot have slashes"};
+    StringView xdg_runtime_dir = getenv("XDG_RUNTIME_DIR");
+    if (xdg_runtime_dir.empty())
+    {
+        // set sticky bit on the shared kakoune directory
+        make_directory(format("{}/kakoune", tmpdir()), 01777);
+        make_directory(format("{}/kakoune/{}", tmpdir(), get_user_name()), 0711);
+        return format("{}/kakoune/{}/{}", tmpdir(), get_user_name(), session);
+    }
+    else
+    {
+        make_directory(format("{}/kakoune", xdg_runtime_dir), 0711);
+        return format("{}/kakoune/{}", xdg_runtime_dir, session);
+    }
+}
+
 static sockaddr_un session_addr(StringView session)
 {
     sockaddr_un addr;
     addr.sun_family = AF_UNIX;
-    auto slash_count = std::count(session.begin(), session.end(), '/');
-    if (slash_count > 1)
-        throw runtime_error{"session names are either <user>/<name> or <name>"};
-    else if (slash_count == 1)
-        format_to(addr.sun_path, "{}/kakoune/{}", tmpdir(), session);
-    else
-        format_to(addr.sun_path, "{}/kakoune/{}/{}", tmpdir(),
-                  get_user_name(), session);
+    strcpy(addr.sun_path, session_path(session).c_str());
```

Comment:
its a shame we have to go through an allocation here now, but I guess its not that bad.